pkgs/servers/matrix-synapse/matrix-appservice-slack/default.nix

```diff
@@ -1,29 +1,44 @@
-{ pkgs, nodejs, stdenv, fetchFromGitHub, lib, ... }:
-let
+{ lib
+, fetchFromGitHub
+, makeWrapper
+, matrix-sdk-crypto-nodejs
+, mkYarnPackage
+, nodejs
+}:
+
+mkYarnPackage rec {
+  pname = "matrix-appservice-slack";
+  version = "2.0.0";
+
   src = fetchFromGitHub {
     owner = "matrix-org";
     repo = "matrix-appservice-slack";
-    rev = "1.11.0";
-    sha256 = "U1EHL1ZwcpCXA9sjya6ry/3Q+gwdQWPUDFN+wp1qjrg=";
+    rev = version;
+    hash = "sha256-iUoZO39v5jvskpF3Rrc0sfeNAbRlqiVa6wFOO8yWH5o=";
   };
+  packageJSON = ./package.json;
+  yarnLock = ./yarn.lock;
+  yarnNix = ./yarn.nix;
```

Comment:
Can this use `fetchYarnDeps` instead? See [here](https://github.com/NixOS/nixpkgs/blob/27a89ba43b0fb735ce867e8ab3d2442f8cc61dad/pkgs/servers/monitoring/matrix-alertmanager/default.nix) for an example, and [here](https://github.com/NixOS/nixpkgs/blob/27a89ba43b0fb735ce867e8ab3d2442f8cc61dad/pkgs/applications/networking/instant-messengers/element/update.sh) for how to get the hash in an update script.

Reply:
done